src/main/java/io/confluent/kafkarest/SimpleConsumerManager.java

```diff
@@ -15,108 +15,83 @@
-    poolInstanceAvailabilityTimeoutMs = config.getInt(KafkaRestConfig.SIMPLE_CONSUMER_POOL_TIMEOUT_MS_CONFIG);
-    time = config.getTime();
 
-    simpleConsumersPools = new ConcurrentHashMap<Broker, SimpleConsumerPool>();
+    int maxPoolSize = config.getInt(KafkaRestConfig.SIMPLE_CONSUMER_MAX_POOL_SIZE_CONFIG);
+    int poolInstanceAvailabilityTimeoutMs = config.getInt(KafkaRestConfig.SIMPLE_CONSUMER_POOL_TIMEOUT_MS_CONFIG);
+    int maxPollTime = config.getInt(KafkaRestConfig.SIMPLE_CONSUMER_MAX_POLL_TIME_CONFIG);
+    Time time = config.getTime();
 
-    // Load decoders
-    Properties props = new Properties();
-    props.setProperty("schema.registry.url",
-        config.getString(KafkaRestConfig.SCHEMA_REGISTRY_URL_CONFIG));
-    avroDecoder = new KafkaAvroDecoder(new VerifiableProperties(props));
+    ConsumerFactory<byte[], byte[]> consumerFactory = simpleConsumerFactory;
 
-    binaryDecoder = new DefaultDecoder(new VerifiableProperties());
+    if (consumerFactory == null) {
+        consumerFactory = new ConsumerFactory<byte[], byte[]>() {
+        private Properties consumerProperties = new Properties();
 
-    jsonDecoder = new KafkaJsonDecoder<Object>(new VerifiableProperties());
-  }
-
-  private SimpleConsumerPool createSimpleConsumerPool() {
-    return new SimpleConsumerPool(maxPoolSize, poolInstanceAvailabilityTimeoutMs, time, simpleConsumerFactory);
-  }
+        {
+          consumerProperties.setProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG,
+              config.getString(KafkaRestConfig.BOOTSTRAP_SERVERS_CONFIG));
+        }
 
-  private SimpleFetcher getSimpleFetcher(final Broker broker) {
-    // When upgrading to Java 1.8, use simpleConsumersPools.computeIfAbsent() instead
-    SimpleConsumerPool pool = simpleConsumersPools.get(broker);
-    if (pool == null) {
-      simpleConsumersPools.putIfAbsent(broker, createSimpleConsumerPool());
-      pool = simpleConsumersPools.get(broker);
+        @Override
+        public Consumer<byte[], byte[]> createConsumer() {
+          return new KafkaConsumer<byte[], byte[]>(consumerProperties,
+              new ByteArrayDeserializer(),
+              new ByteArrayDeserializer());
+        }
+      };
     }
 
-    // TODO: Add support for SSL when simple consumer is changed to use new consumer
-    try {
-      BrokerEndPoint ep = broker.getBrokerEndPoint(SecurityProtocol.PLAINTEXT);
-      return pool.get(ep.host(), ep.port());
-    } catch (BrokerEndPointNotAvailableException e) {
-      throw Errors.noSslSupportException();
-    }
+    simpleConsumersPool =
+      new SimpleConsumerPool(maxPoolSize,
+          poolInstanceAvailabilityTimeoutMs, maxPollTime, time, consumerFactory);
+
+    // Load deserializers
+    Map<String, String> props = new HashMap<>();
+    props.put("schema.registry.url", config.getString(KafkaRestConfig.SCHEMA_REGISTRY_URL_CONFIG));
+    avroDeserializer = new KafkaAvroDeserializer();
+    avroDeserializer.configure(props, true);
```

Comment:
It doesn't have any impact today, but you might want to use separate deserializers for key and value just to make sure this code is future-proof in case the deserializer ever *does* start relying on the `isKey` flag.